### PR TITLE
Prepare Endpoint for Checkins

### LINKED ISSUE #1227: micropub checkins not working
just tested http://ownyourcheckin.wirres.net/ which peso your checkin from 4sq and publish it via micropub back to your site. The status is coming to my known site but not as checkin.
See https://nxd4n.nixekinder.be/2016/checkin-foursquare-4sq-1
This is what is posted:

```
[published] => 2016-02-14T20:04:57+01:00
[location] => geo:50.807987237976,4.313364561688
[place_name] => Georgiana Island
[place_url] => https://foursquare.com/venue/555b4630498eec90b9af983b?ref=YOFY2SHCUPCRRJUBFNI2D1SU3IKOFBHOGSX3ZN5CDEARGODM
[category] => Array
(
[] => checkin
[] => foursquare
[] => 4sq
)

[content] =>
[syndication] => https://foursquare.com/forward/checkin/56c0cfd9498e254d8a1ecb77
```` 
The checkin category is ignored or probably nothing checks against if it is possible to publish a checkin entity if it is enabled.

### PR DESCRIPTION
**Pull request by danito**
This permits to catch checkins from Micropub if mp-type = checkin
resolves #1227

### CODE REVIEW

File: IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php [hunk 1]
```diff
@@ -151,6 +226,13 @@ function post()
                         $this->setInput('like-of', $like_of);
                         $this->setInput('repost-of', $repost_of);
                         $this->setInput('access', 'PUBLIC');
+                        $this->setInput('lat', $lat);
+	                $this->setInput('long', $long);
+        	        $this->setInput('user_address', "");
+                	$this->setInput('placename',$place_name);
+                        $this->setInput('long', $long);
+                        $this->setInput('user_address', "");
+                        $this->setInput('placename',$place_name);
```
Comment (kylewm) [inline IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php]: nit: you've got these three lines twice.

File: IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php [hunk 2]
```diff
@@ -80,6 +81,10 @@ function post()
                 $like_of     = $this->getInput('like-of');
                 $repost_of   = $this->getInput('repost-of');
                 $categories  = $this->getInput('category');
+		$mp_type     = $this->getInput('mp-type');
+		if (!empty($mp_type)) {
+			$type = $mp_type;
+		}
```
Comment (kylewm) [inline IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php]: nit: please use spaces for indentation (4-space indent)

File: IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php [hunk 3]
```diff
@@ -72,6 +72,7 @@ function post()
 
                 // Get details
                 $type        = $this->getInput('h', 'entry');
+                $type        = $this->getInput('h');
```
Comment (kylewm) [inline IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php]: this line repeats the previous one.

File: IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php [hunk 4]
```diff
@@ -117,6 +132,24 @@ function post()
...
+                        $photo_url = $this->getInput('photo');
+                        if ($photo_url) {
+                            $type      = 'photo';
+                            $success   = $this->uploadFromUrl($photo_url);
+                            if (!$success) {
+                            	\Idno\Core\Idno::site()->triggerEvent('indiepub/post/failure', ['page' => $this]);
+                                $this->setResponse(500);
+                                echo "Failed uploading photo from $photo_url";
+                                exit;
+                            }
```
Comment (kylewm) [inline IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php]: this block seems to have migrated down from above (starting around 81 in the old file)? maybe git made a mistake merging?

File: IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php [hunk 5]
```diff
@@ -130,6 +163,48 @@ function post()
 
                 // Get an appropriate plugin, given the content type
                 if ($contentType = ContentType::getRegisteredForIndieWebPostType($type)) {
+                    if ($type == 'photo' && empty($name) && !empty($content)) {
+                        $name    = $content;
+                        $content = '';
+                    }
```
Comment (kylewm) [inline IdnoPlugins/IndiePub/Pages/MicroPub/Endpoint.php]: i'm a little confused what happened here; this partially duplicates code from line ~102

**Comment by kylewm** [comment]
Thank you, by the way, for the PR and for adding this feature!

**Comment by danito** [comment]
I don't know. I merged some old version with new addings. will try to rollback

**Pull request closed without merging**